server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestClusterRerouteAction.java

```diff
@@ -70,6 +77,14 @@ public boolean allowSystemIndexAccessByDefault() {
 
     @Override
     public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
+        if (request.hasParam("metric") == false || Objects.equals(request.param("metric"), "none") == false) {
+            deprecationLogger.critical(
+                DeprecationCategory.API,
+                "reroute_cluster_state",
+                "[state] field is deprecated in a reroute response and will be removed."
```

Comment:
Suggest making this more actionable:

```suggestion
                "The [state] field in the response to the reroute API is deprecated and will be removed in a future version. Specify ?metric=none to adopt the future behaviour."
```